src/d6t/d6t.hpp

```diff
@@ -0,0 +1,280 @@
+             * Get the thermal sensor temperature unit.
+             *
+             * @param unit   Temperature scale unit
+	     *
+             * @return One of the D6T_STATUS_T values
+             */
+            D6T_STATUS_T getTemperatureScale(upm::TemperatureUnit &unit);
+
+            /**
+             * Set the thermal sensor temperature unit.
+             *
+             * @param unit   Temperature scale unit
+	     *
+             * @return One of the D6T_STATUS_T values
+             */
+            D6T_STATUS_T setTemperatureScale(upm::TemperatureUnit unit);
+
+            /**
+             * Display thermal sensor data with PTAT, P0 to P15 and PEC values.
+	     *
+	     * User can call this API at interval greater than or equal to sensor measurement interval.
+             *
+             * @return One of the D6T_STATUS_T values
+             */
+            D6T_STATUS_T displayTemperature();
+
+	    /**
+	     * Set the thermal sensor type.
+	     *
+	     * @param type   Sensor type
+	     *
+	     * @return One of the D6T_STATUS_T values
+	     */
+	    D6T_STATUS_T setSensorType(D6T_SENSOR_TYPES_T type);
+
+	    /**
+	     * Get the thermal sensor type.
+	     *
+	     * @param type   Sensor type
+	     *
+	     * @return One of the D6T_STATUS_T values
+	     */
+	    D6T_STATUS_T getSensorType(D6T_SENSOR_TYPES_T &type);
+
+	    /**
+	     * Get Measurement Interval time.
+	     *
+	     * @param value   Interval value
+	     *
+	     * @return One of the D6T_STATUS_T values
+	     */
+	    D6T_STATUS_T getMeasurementInterval(uint8_t &value);
+
+	    /**
+	     * Set Measurement Interval time.
+	     *
+	     * @param value   Interval value
+	     *
+	     * @return One of the D6T_STATUS_T values
+	     */
+	    D6T_STATUS_T setMeasurementInterval(uint8_t value);
+
+
+        private:
+            int m_bus;
+            int m_address;
+	    upm::TemperatureUnit m_tempScale;
+	    D6T_SENSOR_TYPES_T d6t_sensor_type;
+	    D6T_SENSOR_BUFFER_LENGTH_T d6t_sensor_data_length;
+	    uint8_t *m_buf = NULL;
+            mraa::I2c* m_i2c;
+	    static uint8_t intervalValues1Channel[D6T_1AND8_CHANNEL_VALUE_BUFFER_ROW_LENGTH][D6T_1AND8_CHANNEL_VALUE_BUFFER_COLUMN_LENGTH];
+            static uint8_t intervalValues8Channel[D6T_1AND8_CHANNEL_VALUE_BUFFER_ROW_LENGTH][D6T_1AND8_CHANNEL_VALUE_BUFFER_COLUMN_LENGTH];
+            static uint8_t intervalValues16Channel[D6T_16_CHANNEL_VALUE_BUFFER_ROW_LENGTH][D6T_16_CHANNEL_VALUE_BUFFER_COLUMN_LENGTH];
+	    static uint8_t intervalBytePattern1And8Channel[D6T_1AND8_CHANNEL_REG_PATTERN_ROW_LENGTH][D6T_1AND8_CHANNEL_REG_PATTERN_COLUMN_LENGTH];
+	    static uint8_t intervalBytePattern16Channel[D6T_16_CHANNEL_REG_PATTERN_ROW_LENGTH][D6T_16_CHANNEL_REG_PATTERN_COLUMN_LENGTH];
+
+            /**
+            * Returns the temperature. This method is not publicly accessible.
+            *
+            * @return Dummy value
+            */
+            virtual float getTemperature(){return 0.0;}
```

Comment:
Would be preferred not to hide the default interface call if possible, but it's understandable if it makes no sense for your sensor. Have you thought about maybe having this return the average of the sensor's zones? I'm thinking that would make for a quick and easy way to test if these devices power on correctly.

Reply:
Currently, we are not thinking in this direction.